cmake/external/curl.cmake

```diff
@@ -34,6 +34,17 @@ if (NOT TARGET curl_project)
         set(PARALLEL "")
     endif ()
 
+    # When CMake passes the configure command to the shell, the semi-colon will
+    # also be interpreted by the shell so it needs to be escaped. Quoting does
+    # not work, so instead we replace the semi-colon with a different separator
+    # and specify that using LIST_SEPARATOR below. This ensures that our RPATH
+    # will contain both directories.
+    set(GOOGLE_CLOUD_CPP_INSTALL_RPATH "<INSTALL_DIR>/lib;<INSTALL_DIR>/lib64")
+    string(REPLACE ";"
+                   "|"
```

Comment:
Is there are reason not to use ":", which is a typical path-list separator and also not a shell meta-character.

Reply:
Not sure if the actual delimiter matters. CMake will turn `-DCMAKE_INSTALL_RPATH=<INSTALL_DIR>/lib|<INSTALL_DIR>/lib64` into `-DCMAKE_INSTALL_RPATH=/v/build-output/fedora-29-clang-Debug/external/lib\;/v/build-output/fedora-29-clang-Debug/external/lib64;-GUnix Makefiles` so unless you're viewing the logs it won't matter. Also, my comment may be incorrect since the transformation is done on the CMake side. I'll update.

Reply:
So I reran on master to see what exactly CMake was doing. The problem is that it doesn't escape the semi-colon if you use a semi-colon, i.e: `-DCMAKE_INSTALL_RPATH=<INSTALL_DIR>/lib;<INSTALL_DIR>/lib64` becomes `-DCMAKE_INSTALL_RPATH=/v/build-output/fedora-29-clang-Debug/external/lib;/v/build-output/fedora-29-clang-Debug/external/lib64`. This then is interpreted as a literal `;` by the shell.

Reply:
If the actual delimiter doesn't matter, I'd certainly vote for ":".

But let me ask another question: Why specify the list using ";" and then replace ";" with "|" rather than just using "|" in the first place?

Reply:
Because we want the paths to be separate arguments. I will look more into this later, though.

Reply:
So, it turns out you can't do `LIST_SEPARATOR :`, because it's a special character for CMake:

```
CMake Error at /usr/share/cmake/Modules/ExternalProject.cmake:1944 (file):
  Error evaluating generator expression:

    $<BOOL\;ON>
```

Reply:
@devbww I think we prefer using `;` because that character is the magic list separator in CMake.  All lists get converted to a string with the elements separated by `;`.  Nobody ever said that `CMake` had the best designed programming language.

Reply:
The necessity for this workaround and the bug report I linked to in the PR description suggests it may actually be an outstanding bug in CMake. It may be worth reporting in the gitlab issue tracker.

Reply:
Ack.  Thanks.